lib/phonebook/tools/function.dart

```diff
@@ -16,3 +22,42 @@ String nameConstructor(Map<String, AsyncValue<List<bool>>> data) {
   }
   return name.substring(0, name.length - 1);
 }
+
+List<CompleteMember> sortMembers(List<CompleteMember> members,
+    String associationId, List<String> rolesTags) {
+  List<CompleteMember> sorted = [];
+  List<List<CompleteMember>> sortedByRole =
+      List.generate(rolesTags.length, (index) => []);
+  for (CompleteMember member in members) {
+    Membership membership = member.memberships
+        .firstWhere((element) => element.associationId == associationId);
+    List<String> memberRoleTags = membership.rolesTags;
+    int highestPosition = 1000;
+    for (String roleTag in memberRoleTags) {
+      int position = rolesTags.indexOf(roleTag);
+      if (position < highestPosition) {
+        highestPosition = position;
+      }
+    }
+    sortedByRole[highestPosition].add(member);
+  }
+  for (List<CompleteMember> list in sortedByRole) {
+    sorted.addAll(list);
+  }
```

Comment:
I think you can sort using a one-liner and an auxiliary function, it may be easier to understand : 

`int getPosition(CompleteMember member, String associationId, List<String> rolesTags) {
    Membership membership = member.memberships.firstWhere((element) => element.associationId == associationId);
    return min(membership.rolesTags.map((roleTag) => rolesTags.indexOf(roleTag));
}`
And then
`sorted = members.sort((a, b) => getPosition(a).compareTo(getPosition(b));`